File: SWEA/D2/1859. 백만 장자 프로젝트/백만 장자 프로젝트.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
 
class Solution
{
    public static void main(String args[]) throws Exception
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());
        for(int test_case = 1; test_case <= T; test_case++)
        {
            int N = Integer.parseInt(br.readLine());
            long[] price = new long[N];
            StringTokenizer st = new StringTokenizer(br.readLine());
            for(int i=0; i<N; i++) price[i] = Integer.parseInt(st.nextToken());
            long sum=0;
            long curMax=price[N-1];
            for(int i=N-2; i>=0; i--) {
                if(price[i]>curMax) curMax=price[i];
                else sum+=(curMax-price[i]);
            }
            System.out.println("#"+test_case+" "+sum);
        }
    }
}

File: SWEA/D2/1859. 백만 장자 프로젝트/백만 장자 프로젝트.cpp
#include <bits/stdc++.h>
using namespace std;
int N;
long arr[1000001];
 
int main(int argc, char** argv)
{
    int test_case;
    int T;
    cin>>T;
    for(test_case = 1; test_case <= T; ++test_case)
    {
        cin >> N;
        for(int i=1; i<=N; i++) cin >> arr[i];
        long sum=0;
        long curMax=arr[N];
        for(int i=N-1; i>=1; i--) {
            if(curMax<arr[i]) curMax=arr[i];
            else sum+=curMax-arr[i];
        }
        cout << "#" << test_case << " " << sum << '\n';
    }
    return 0;
}